File: app/src/main/java/com/tregula/graedukacyjna/domain/database/dao/ContinentDao.java
package com.tregula.graedukacyjna.domain.database.dao;

import com.tregula.graedukacyjna.domain.database.entity.Continent;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import io.reactivex.Single;

@Dao
public interface ContinentDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(Continent continent);

    @Query("SELECT * from continents")
    Single<List<Continent>> getContinents();

    @Query("DELETE FROM continents")
    void removeContinents();
}

File: app/src/main/java/com/tregula/graedukacyjna/domain/database/dao/ContinentDao.kt
package com.tregula.graedukacyjna.domain.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.tregula.graedukacyjna.domain.database.entity.Continent
import io.reactivex.Single

@Dao
interface ContinentDao {

    @Query("SELECT * from continents")
    fun getContinents(): Single<List<Continent>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(continent: Continent): Long

    @Query("DELETE FROM continents")
    fun removeContinents()
}
